Tasks/VsTest/distributedtest.ts

```diff
@@ -143,11 +143,36 @@ export class DistributedTest {
             throw new Error('Failed to start Modules/DTAExecutionHost.exe.');
         });
 
+        var consolidatedCiData = {
+                    agentFailure: false,
+                    batchingType: models.BatchingType[this.dtaTestConfig.batchingType],
+                    batchsize: this.dtaTestConfig.numberOfTestCasesPerSlice,
+                    codeCoverageEnabled: this.dtaTestConfig.codeCoverageEnabled,
+                    customSlicingEnabled: utils.Helper.isNullOrUndefined(this.dtaTestConfig.customSlicingenabled) ? 'false' : 'true',
+                    environmenturi: this.dtaTestConfig.dtaEnvironment.environmentUri, 
+                    numberOfAgentsInPhase: this.dtaTestConfig.numberOfAgentsInPhase,
+                    numberOfTestCasesPerSlice: this.dtaTestConfig.numberOfTestCasesPerSlice,
+                    overridesTestRunParameters: utils.Helper.isNullOrUndefined(this.dtaTestConfig.overrideTestrunParameters) ? 'false' : 'true',
+                    pathToCustomTestAdaptersSet: utils.Helper.isNullOrUndefined(this.dtaTestConfig.pathtoCustomTestAdapters) ? 'false' : 'true',
+                    pipeline: tl.getVariable('release.releaseUri') != null ? "release" : "build",
+                    runInIsolation: this.dtaTestConfig.runTestsInIsolation,
+                    task: "VsTest",
+                    testSelection: this.dtaTestConfig.testSelection,
+                    usesMultiCoreParallel: this.dtaTestConfig.runInParallel,
+                    usesTia: this.dtaTestConfig.tiaConfig.tiaEnabled,
+                    useVsTestConsole: this.dtaTestConfig.useVsTestConsole,
+                    videoCoverageEnabled: utils.Helper.isNullOrUndefined(this.dtaTestConfig.videoCoverageEnabled) ? 'false' : 'true',
+                    vsTestVersion: this.dtaTestConfig.vsTestVersionDetails.majorVersion + '.' + this.dtaTestConfig.vsTestVersionDetails.minorversion + '.' + this.dtaTestConfig.vsTestVersionDetails.patchNumber
+                };
+
         proc.on('close', (code) => {
             if (code !== 0) {
                 tl.debug('Modules/DTAExecutionHost.exe process exited with code ' + code);
+                consolidatedCiData.agentFailure = true;
+                ci.publishEvent(consolidatedCiData);
             } else {
                 tl.debug('Modules/DTAExecutionHost.exe exited');
+                ci.publishEvent(consolidatedCiData);
```

Comment:
or call this code after the if/else block using single call.

Reply:
hmmm. okay